projects/igniteui-angular/src/lib/core/styles/components/chips/_chips-theme.scss

```diff
@@ -0,0 +1,245 @@
+        } @else {
+            $chip-hover-background: darken($chip-background, 12%);
+
+            @if not($text-color) and $chip-hover-background{
+                $text-color: text-contrast($chip-hover-background);
+            }
+
+            @if not($order-icon-color) and $chip-hover-background{
+                $order-icon-color: text-contrast($chip-hover-background);
+            }
+
+            @if not($remove-icon-color) and $chip-hover-background{
+                $remove-icon-color: rgba(text-contrast($chip-hover-background), .4);
+            }
+
+            @if not($remove-icon-hover-color) and $chip-hover-background{
+                $remove-icon-hover-color: rgba(text-contrast($chip-hover-background), 1);
+            }
+        }
+    }
+
+    @if not($text-hover-color) and $chip-hover-background{
+        $text-hover-color: text-contrast($chip-hover-background);
+    }
+
+    @if not($remove-icon-hover-color) and $chip-hover-background{
+        $remove-icon-hover-color: rgba(text-contrast($chip-hover-background), .4);
+    }
+
+    @if not($remove-icon-color) and $chip-hover-background{
+        $remove-icon-color: rgba(text-contrast($chip-hover-background), .4);
+    }
+
+    @if not($order-icon-hover-color) and $chip-hover-background{
+        $order-icon-hover-color: text-contrast($chip-hover-background);
+    }
+
+
+    @return extend($default-theme, (
+        palette: $palette,
+        chip-background: $chip-background,
+        chip-hover-background: $chip-hover-background,
+
+        text-color: $text-color,
+        text-hover-color: $text-hover-color,
+        order-icon-hover-color: $order-icon-hover-color,
+
+        order-icon-color: $order-icon-color,
+        dir-icon-color: $dir-icon-color,
+        remove-icon-color: $remove-icon-color,
+        remove-icon-hover-color: $remove-icon-hover-color,
+    ));
+}
+
+/// @param {Map} $theme - The theme used to style the component.
+/// @requires {mixin} igx-root-css-vars
+/// @requires rem
+/// @requires var
+@mixin igx-chips($theme) {
+    @include igx-root-css-vars($theme);
+
+    %igx-chips-area,
+    %igx-chips-area__item,
+    %igx-chips-area__label,
+    %igx-chips-area__chip-connector {
+        display: flex;
+        align-items: center;
+    }
+
+    %igx-chips-area {
+        justify-content: flex-start;
+        width: 100%;
+    }
+
+    %igx-chips-area__item {
+        user-select: none;
+        cursor: pointer;
+        border-radius: rem(17px);
+        padding: rem(4) rem(16);
+        height: rem(34px);
+        transition: all 120ms ease-in-out;
+        background: --var($theme, 'chip-background');
+
+        &:hover,
+        &:focus{
+            background: --var($theme, 'chip-hover-background');
+
+            %igx-chips-area__text {
+                color: --var($theme, 'text-hover-color');
+            }
+
+            transition: all 120ms ease-in-out;
+
+            .igx-icon {
+                transition: all 120ms ease-in-out;
```

Comment:
I see this repeated a lot. This is not great for maintainability. Please cache in a variable and use timing functions from the predefined in the animations utils library.